README.adoc

```diff
@@ -1,15 +1,62 @@
 = keep-core
 
-The core code behind the https://keep.network[Keep network].
+https://circleci.com/gh/keep-network/keep-core[image:https://circleci.com/gh/keep-network/keep-core.svg?style=svg&circle-token=ec728f5ca814b6cb2db5ffeb7258151b752a207e[CircleCI Build Status]]
+https://discord.gg/wYezN7v[image:https://img.shields.io/badge/chat-Discord-blueViolet.svg[Chat with us on Discord]]
 
-== Getting Started as a Developer
+The core contracts and reference client implementation behind the Keep network,
+a privacy, interoperability, and censorship-resistance toolkit for developers
+on Ethereum.
 
-Start with the README in link:docs/development/[the developer docs directory].
+== What's a keep?
 
-== Directory structure
+The network offers application developers **keeps**, small off-chain data
+containers for private storage and computation that can be opened, closed, and
+managed by smart contracts autonomously.
 
-The directory structure used in the Keep repository is very similar to that used
-in other Go projects:
+Keeps are maintained by stakers, actors who run nodes and have skin in the
+game. When a new keep is opened, the requisite number of stakers are chosen
+via a BLS-based random beacon to maintain the keep.
+
+The first type of keep launching with the network is the `BondedECDSAKeep`,
+allowing smart contracts to generate private keys and sign messages without
+endangering key material. ECDSA keeps mean decentralized signing, cross-chain
+applications, and new tools for custodial applications — from Solidity. This
+capability is used heavily by https://tbtc.network/[tBTC].
+
+== Getting Started
+
+A good place to start is the link:docs/[the docs directory].
+
+=== Running a Node
+
+To run your own node in the Keep network, follow the
+link:docs/keep-client-quickstart.adoc[`keep-client` quickstart]. Feedback on
+this process and the documentation
+https://github.com/keep-network/keep-core/issues[is appreciated!]
+
+=== dApp Developers
+
+dApp developers will be most interested in the smart contracts exposing Keep's
+on-chain facilities.
+
+The contracts can be found in the link:solidity/[`solidity/`] directory. They
+can be used to request miner-resistant random numbers, as well as handle
+creating and managing keeps, generating new secrets, and requesting signatures.
+
+It runs the Keep network’s random beacon and participates in related
+computations.
+
+=== Client Developers
+
+Client developers will be most interested in the link:go/[reference Keep Go
+client], as well as the link:docs/rfc/[RFCs],
```

Comment:
Also `go/` is not now nor has it ever been a thing. The directory structure is explained below :wink:

Reply:
> nor has it ever

Turns out [this isn't true](https://github.com/keep-network/keep-core/tree/9635be6899ae2e76bdeb668ea591c73df3288e98), for those keeping track :grin: Had a gnawing sense of overcommitment there.

Reply:
I think the move here is pointing folks to the directory structure at the end of the doc + the RFCs.

Reply:
Heh the `go/` came from a very very old version of master (my fork) then rebased. Thanks, will fix

Reply:
Omg so old 😂

Reply:
Yeah it was sort of depressing to realize :sweat_smile: